src/features/broker-protection/actions/build-url-transforms.js

```diff
@@ -114,13 +114,29 @@ function processPathname (pathname, action, userData) {
  *  - `name` would be processed with the 'lowercase' transform
  *  - `age` would be used without processing
  *
+ * The regular expression `/\$%7B(.+?)%7D|\$\{(.+?)}/g` is designed to match and capture
+ * the content within template literals in two formats: encoded and plain.
+ *
+ * 1. Encoded Format: `\$%7B(.+?)%7D`
+ *    - Matches encoded template strings that start with `$%7B` and end with `%7D`.
+ *    - `(.+?)` is a non-greedy capture group that matches any character sequence
+ *      between `$%7B` and `%7D`, capturing the content of the encoded template literal.
+ *
+ * 2. Plain Format: `\$\{(.+?)\}`
+ *    - Matches plain template strings that start with `${` and end with `}`.
+ *    - `(.+?)` here again is a non-greedy capture group that matches any character
+ *      sequence between `${` and `}`, capturing the content of the plain template literal.
+ *
+ * This regular expression is used to identify and process these template literals within the input string,
+ * allowing the function to replace them with corresponding data from `userData` after applying any specified transformations.
+ *
  * @param {string} input
  * @param {BuildUrlAction} action
  * @param {Record<string, string|number>} userData
  */
-export function processOne (input, action, userData) {
-    return String(input).replace(/\$%7B(.+?)%7D|\$\{(.+?)}/g, (match, value) => {
-        const comparison = value || match.slice(2, -1)
+export function processTemplateStringWithUserData (input, action, userData) {
+    return String(input).replace(/\$%7B(.+?)%7D|\$\{(.+?)}/g, (match, encodedValue, plainValue) => {
+        const comparison = encodedValue ?? plainValue
```

Comment:
do we need to decode `encodedValue`?

Reply:
Ahh sorry I missed this!

We don't need to, anything that gets this far has already been through the URL constructor, which decodes.

Reply:
wait, why do we need the `%7B` and `%7D` bits in the regex then?

Reply:
This is because pathname and query segments are treated differently when passed through the URL constructor, so this handles both situations.

if this causes confusion, we could split it up, and make it clear which we're processing.